### PR TITLE
JAX-WS and JAX-RS 2.1: Add Java 23 and above compatible Subject methods in DefaultLogEventMapper

### PR DESCRIPTION
**Pull request by neuwerk**
This PR adds support for Java 23 in Apache CXF's `DefaultLogEventMapper` by modifying how `Subject` is called depending on the Java level.

### CODE REVIEW

**Comment by neuwerk** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.jaxws.2.2.webcontainer_fat, com.ibm.ws.jaxws.2.2.webcontainer_fat_extended, com.ibm.ws.jaxws.cdi_fat, com.ibm.ws.jaxws.clientcontainer_fat, com.ibm.ws.jaxws.ejb_fat, io.openliberty.jaxrs.2.1.jaxws.2.2_fat, io.openliberty.jaxws.beanvalidation_fat, io.openliberty.jaxws.executor_fat, io.openliberty.jaxws.global.handler.internal_fat, io.openliberty.jaxws.security_fat, io.openliberty.jaxws.security_fat.1, io.openliberty.jaxws.security_fat.ssl, io.openliberty.jaxws.tools_fat, io.openliberty.jaxws.wsat_fat, io.openliberty.jaxws_fat, com.ibm.ws.wsat.common_fat, com.ibm.ws.wsat.concurrent_fat, com.ibm.ws.wsat.migration_fat.1, com.ibm.ws.wsat.migration_fat.2, com.ibm.ws.wsat.migration_fat.3, com.ibm.ws.wsat.migration_fat.4, com.ibm.ws.wsat.migration_fat.5, com.ibm.ws.wsat.migration_fat.6, com.ibm.ws.wsat.migration_fat.7, com.ibm.ws.wsat.recovery_fat.lps, com.ibm.ws.wsat.recovery_fat.multi.1, com.ibm.ws.wsat.recovery_fat.multi.2, com.ibm.ws.wsat.recovery_fat.multi.3, com.ibm.ws.wsat.recovery_fat.multi.4, com.ibm.ws.wsat.recovery_fat.multi.5, com.ibm.ws.wsat.recovery_fat.multi.6, com.ibm.ws.wsat.recovery_fat.multi.7, com.ibm.ws.wsat.recovery_fat.single.1, com.ibm.ws.wsat.recovery_fat.single.2, com.ibm.ws.wsat_fat.assertion, com.ibm.ws.wsat_fat.db, com.ibm.ws.wsat_fat.db_optional, com.ibm.ws.wsat_fat.dbs, com.ibm.ws.wsat_fat.dbs_optional, com.ibm.ws.wsat_fat.multi, com.ibm.ws.wsat_fat.multi.misroute, com.ibm.ws.wsat_fat.ssl, com.ibm.ws.wsat_fat.ssl.misroute, io.openliberty.jaxws.wsat_fat, com.ibm.ws.wssecurity_fat.wsscxf.1, com.ibm.ws.wssecurity_fat.wsscxf.2, com.ibm.ws.wssecurity_fat.wsscxf.3, com.ibm.ws.wssecurity_fat.wsscxf.4, com.ibm.ws.wssecurity_fat.wsscxf.saml.1, com.ibm.ws.wssecurity_fat.wsscxf.saml.2, com.ibm.ws.wssecurity_fat.wsscxf.saml.3, com.ibm.ws.wssecurity_fat.wsscxf.saml.5, com.ibm.ws.wssecurity_fat.wsscxf.saml.6, io.openliberty.xmlws.4.0.internal_fat, com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.0.cdi.1.2_fat,com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.1_fat,com.ibm.ws.jaxrs.2.1_fat_executor,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1.client_fat,com.ibm.ws.jaxrs.2.1.sse_fat,com.ibm.ws.jaxrs.2.x_fat_clientProps, io.openliberty.restfulWS.3.0_fat,io.openliberty.restfulWS.3.0.cdi.3.0_fat,io.openliberty.restfulWS.3.0.client_fat,io.openliberty.restfulWS.3.1_fat

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.features.logging.3.2/bnd.bnd [hunk 1]
```diff
@@ -16,7 +16,11 @@ cxfVersion=3.5.5
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.features.logging.3.2/bnd.bnd:16]: Change date to 2024

**Review by WhiteCat22 — DISMISSED** [review]

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.features.logging.3.2/src/org/apache/cxf/ext/logging/event/DefaultLogEventMapper.java [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+            int majorVersion = javaVersion.indexOf(".");
+            
+            if(majorVersion != -1) { 
+                javaVersion = javaVersion.substring(0, majorVersion);
+            }
+        } 
+        
+        return Integer.parseInt(javaVersion);
+    }
+    // Liberty Change End
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.features.logging.3.2/src/org/apache/cxf/ext/logging/event/DefaultLogEventMapper.java]:
```suggestion
```

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.features.logging.3.2/src/org/apache/cxf/ext/logging/event/DefaultLogEventMapper.java [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+     * So for versions before Java 23, we can still use Subject.getSubject(), but for Java 23 and beyond,
+     * we need to switch to using Subject.current(), which was introduced in Java 18.
+     *
+     * So for Java 22 and earlier, this returns Subject.getSubject(...)
+     * For Java 23 and later, this method returns Subject.current()
+     *
+     * @return
+     */
+    public static Subject getCurrentSubject() {
+        if (JAVA_VERSION <= 22) {
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.features.logging.3.2/src/org/apache/cxf/ext/logging/event/DefaultLogEventMapper.java]:
```suggestion
        if (JavaInfo.majorVersion() <= 22) {
```

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.features.logging.3.2/src/org/apache/cxf/ext/logging/event/DefaultLogEventMapper.java [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.logging.Logger;
+
+import javax.security.auth.Subject;
+
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.features.logging.3.2/src/org/apache/cxf/ext/logging/event/DefaultLogEventMapper.java:36]:
```suggestion

import com.ibm.ws.kernel.service.util.JavaInfo;

```

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.features.logging.3.2/bnd.bnd [hunk 5]
```diff
@@ -16,7 +16,11 @@ cxfVersion=3.5.5
...
   @${repo;org.apache.cxf:cxf-rt-features-logging;${cxfVersion};EXACT}!/!*-INF/*,\
-  META-INF/cxf=resources/META-INF/cxf
+  META-INF/cxf=resources/META-INF/cxf, \
+  org/apache/cxf=${bin}/org/apache/cxf
+
+
+instrument.disabled: true
 
 -buildpath: \
   org.apache.cxf:cxf-rt-features-logging;strategy=exact;version=${cxfVersion}, \
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.features.logging.3.2/bnd.bnd:25]:
```suggestion
  org.apache.cxf:cxf-rt-features-logging;strategy=exact;version=${cxfVersion}, \
  com.ibm.ws.kernel.service;version=latest, \
```

**Review by jim-krueger — DISMISSED** [review]

**Comment by neuwerk** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.jaxws.2.2.webcontainer_fat, com.ibm.ws.jaxws.2.2.webcontainer_fat_extended, com.ibm.ws.jaxws.cdi_fat, com.ibm.ws.jaxws.clientcontainer_fat, com.ibm.ws.jaxws.ejb_fat, io.openliberty.jaxrs.2.1.jaxws.2.2_fat, io.openliberty.jaxws.beanvalidation_fat, io.openliberty.jaxws.executor_fat, io.openliberty.jaxws.global.handler.internal_fat, io.openliberty.jaxws.security_fat, io.openliberty.jaxws.security_fat.1, io.openliberty.jaxws.security_fat.ssl, io.openliberty.jaxws.tools_fat, io.openliberty.jaxws.wsat_fat, io.openliberty.jaxws_fat, com.ibm.ws.wsat.common_fat, com.ibm.ws.wsat.concurrent_fat, com.ibm.ws.wsat.migration_fat.1, com.ibm.ws.wsat.migration_fat.2, com.ibm.ws.wsat.migration_fat.3, com.ibm.ws.wsat.migration_fat.4, com.ibm.ws.wsat.migration_fat.5, com.ibm.ws.wsat.migration_fat.6, com.ibm.ws.wsat.migration_fat.7, com.ibm.ws.wsat.recovery_fat.lps, com.ibm.ws.wsat.recovery_fat.multi.1, com.ibm.ws.wsat.recovery_fat.multi.2, com.ibm.ws.wsat.recovery_fat.multi.3, com.ibm.ws.wsat.recovery_fat.multi.4, com.ibm.ws.wsat.recovery_fat.multi.5, com.ibm.ws.wsat.recovery_fat.multi.6, com.ibm.ws.wsat.recovery_fat.multi.7, com.ibm.ws.wsat.recovery_fat.single.1, com.ibm.ws.wsat.recovery_fat.single.2, com.ibm.ws.wsat_fat.assertion, com.ibm.ws.wsat_fat.db, com.ibm.ws.wsat_fat.db_optional, com.ibm.ws.wsat_fat.dbs, com.ibm.ws.wsat_fat.dbs_optional, com.ibm.ws.wsat_fat.multi, com.ibm.ws.wsat_fat.multi.misroute, com.ibm.ws.wsat_fat.ssl, com.ibm.ws.wsat_fat.ssl.misroute, io.openliberty.jaxws.wsat_fat, com.ibm.ws.wssecurity_fat.wsscxf.1, com.ibm.ws.wssecurity_fat.wsscxf.2, com.ibm.ws.wssecurity_fat.wsscxf.3, com.ibm.ws.wssecurity_fat.wsscxf.4, com.ibm.ws.wssecurity_fat.wsscxf.saml.1, com.ibm.ws.wssecurity_fat.wsscxf.saml.2, com.ibm.ws.wssecurity_fat.wsscxf.saml.3, com.ibm.ws.wssecurity_fat.wsscxf.saml.5, com.ibm.ws.wssecurity_fat.wsscxf.saml.6, io.openliberty.xmlws.4.0.internal_fat, com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.0.cdi.1.2_fat,com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.1_fat,com.ibm.ws.jaxrs.2.1_fat_executor,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1.client_fat,com.ibm.ws.jaxrs.2.1.sse_fat,com.ibm.ws.jaxrs.2.x_fat_clientProps, io.openliberty.restfulWS.3.0_fat,io.openliberty.restfulWS.3.0.cdi.3.0_fat,io.openliberty.restfulWS.3.0.client_fat,io.openliberty.restfulWS.3.1_fat

**Comment by neuwerk** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.jaxws.2.2.webcontainer_fat, com.ibm.ws.jaxws.2.2.webcontainer_fat_extended, com.ibm.ws.jaxws.cdi_fat, com.ibm.ws.jaxws.clientcontainer_fat, com.ibm.ws.jaxws.ejb_fat, io.openliberty.jaxrs.2.1.jaxws.2.2_fat, io.openliberty.jaxws.beanvalidation_fat, io.openliberty.jaxws.executor_fat, io.openliberty.jaxws.global.handler.internal_fat, io.openliberty.jaxws.security_fat, io.openliberty.jaxws.security_fat.1, io.openliberty.jaxws.security_fat.ssl, io.openliberty.jaxws.tools_fat, io.openliberty.jaxws.wsat_fat, io.openliberty.jaxws_fat, com.ibm.ws.wsat.common_fat, com.ibm.ws.wsat.concurrent_fat, com.ibm.ws.wsat.migration_fat.1, com.ibm.ws.wsat.migration_fat.2, com.ibm.ws.wsat.migration_fat.3, com.ibm.ws.wsat.migration_fat.4, com.ibm.ws.wsat.migration_fat.5, com.ibm.ws.wsat.migration_fat.6, com.ibm.ws.wsat.migration_fat.7, com.ibm.ws.wsat.recovery_fat.lps, com.ibm.ws.wsat.recovery_fat.multi.1, com.ibm.ws.wsat.recovery_fat.multi.2, com.ibm.ws.wsat.recovery_fat.multi.3, com.ibm.ws.wsat.recovery_fat.multi.4, com.ibm.ws.wsat.recovery_fat.multi.5, com.ibm.ws.wsat.recovery_fat.multi.6, com.ibm.ws.wsat.recovery_fat.multi.7, com.ibm.ws.wsat.recovery_fat.single.1, com.ibm.ws.wsat.recovery_fat.single.2, com.ibm.ws.wsat_fat.assertion, com.ibm.ws.wsat_fat.db, com.ibm.ws.wsat_fat.db_optional, com.ibm.ws.wsat_fat.dbs, com.ibm.ws.wsat_fat.dbs_optional, com.ibm.ws.wsat_fat.multi, com.ibm.ws.wsat_fat.multi.misroute, com.ibm.ws.wsat_fat.ssl, com.ibm.ws.wsat_fat.ssl.misroute, io.openliberty.jaxws.wsat_fat, com.ibm.ws.wssecurity_fat.wsscxf.1, com.ibm.ws.wssecurity_fat.wsscxf.2, com.ibm.ws.wssecurity_fat.wsscxf.3, com.ibm.ws.wssecurity_fat.wsscxf.4, com.ibm.ws.wssecurity_fat.wsscxf.saml.1, com.ibm.ws.wssecurity_fat.wsscxf.saml.2, com.ibm.ws.wssecurity_fat.wsscxf.saml.3, com.ibm.ws.wssecurity_fat.wsscxf.saml.5, com.ibm.ws.wssecurity_fat.wsscxf.saml.6, io.openliberty.xmlws.4.0.internal_fat, com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.0.cdi.1.2_fat,com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.1_fat,com.ibm.ws.jaxrs.2.1_fat_executor,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1.client_fat,com.ibm.ws.jaxrs.2.1.sse_fat,com.ibm.ws.jaxrs.2.x_fat_clientProps, io.openliberty.restfulWS.3.0_fat,io.openliberty.restfulWS.3.0.cdi.3.0_fat,io.openliberty.restfulWS.3.0.client_fat,io.openliberty.restfulWS.3.1_fat

**Review by WhiteCat22 — APPROVED** [review]

**Review by jim-krueger — APPROVED** [review]

**Comment by jhanders34** [comment]
#libby

**Pull request merged**